app/services/catalog/override_icon.rb

```diff
@@ -0,0 +1,18 @@
+module Catalog
+  class OverrideIcon
+    attr_reader :icon
+
+    def initialize(icon_id, portfolio_item_id)
+      @icon = Icon.find(icon_id)
+      @portfolio_item = PortfolioItem.find(portfolio_item_id)
+    end
+
+    def process
+      # there should only ever be one icon, the others are discarded.
+      Catalog::SoftDelete.new(@portfolio_item.icons.first).process
+      @portfolio_item.icons << @icon
```

Comment:
Should we just be calling `@portfolio_item.add_icon(@icon)`? Also do we not need to call `.save`?

Reply:
Yeah I like that idea, I'll change that. We don't need to call save because when we manually manipulate the relation (`=` for `has_one`, `>>` for `has_many`) it automatically does it for us.